tests/cypress/integration/actions_objects2/case_108_rotated_bounding_boxes.js

```diff
@@ -0,0 +1,129 @@
+            cy.get('#cvat_canvas_shape_2').then((shape2) => {
+                const shapeAttrs = shape2.attr('transform').split('(')[1].split(')')[0].split(',');
+                for (const i of shapeAttrs) {
+                    transformMatrixShape.push(Number(i).toFixed(3));
+                }
+            });
+        });
+
+        it('Check interpolation, merging/splitting rotated shapes.', () => {
+            // Check track roration on all frames
+            for (let frame = 1; frame < 10; frame++) {
+                cy.goToNextFrame(frame);
+                cy.get('#cvat_canvas_shape_2').should('have.attr', 'transform').then(($transform) => {
+                    const transform = $transform.split('(')[1].split(')')[0].split(',');
+                    const transformMatrix = [];
+                    for (const i of transform) {
+                        transformMatrix.push(Number(i).toFixed(3));
+                    }
+                    expect(transformMatrix).to.deep.eq(transformMatrixShape);
+                });
+            }
+            // Merge tracks
+            cy.get('.cvat-merge-control').click();
+            cy.get('#cvat_canvas_shape_2').click();
+            cy.get('body').type('m');
+
+            testShapeRotate('#cvat_canvas_shape_2', 350, 250, '91.9°');
+
+            // Comparison of the values of the shape attribute of the current frame with the previous frame
+            for (let frame = 8; frame >= 1; frame--) {
+                cy.get('#cvat_canvas_shape_2').should('have.attr', 'transform').then(($transform) => {
+                    const transform = $transform.split('(')[1].split(')')[0].split(',');
+                    const transformMatrix = [];
+                    for (const i of transform) {
+                        transformMatrix.push(Number(i).toFixed(3));
+                    }
+                    cy.goToPreviousFrame(frame);
+                    cy.get('#cvat_canvas_shape_2').should('have.attr', 'transform').then(($transform2) => {
+                        const transform2 = $transform2.split('(')[1].split(')')[0].split(',');
+                        const transformMatrix2 = [];
+                        for (const i of transform2) {
+                            transformMatrix2.push(Number(i).toFixed(3));
+                        }
+                        expect(transformMatrix).not.deep.eq(transformMatrix2);
+                    });
+                });
+            }
+            cy.goCheckFrameNumber(2);
+
+            // Split tracks
+            cy.get('.cvat-split-track-control').click();
+            // A single click does not reproduce the split a track scenario in cypress test.
+            cy.get('#cvat_canvas_shape_2').click().click();
+            cy.get('#cvat_canvas_shape_3').should('have.attr', 'transform').then((shapeTransform) => {
+                cy.get('#cvat_canvas_shape_4').should('have.attr', 'transform', shapeTransform);
```

Comment:
One of shapes (3 or 4) should be invisible. Is it correct to try comparing their attributes?

Reply:
This is probably normal. It is not necessary to make the shape visible. Or is another check expected here?